docs/guides/authentication.md

```diff
@@ -0,0 +1,76 @@
+
+The API key is in reality a permanent refresh token, meaning you cannot use it directly to authenticate API calls. You first need to exchange it for an access token, follow the [Authenticate API calls](/guides/authentication#authenticate-api-calls) section for detailed steps.
+
+::: info
+Make sure to copy the generated key and store it in a safe place, as it will not be possible to see it again.
+:::
+
+## Discord OAuth2
+
+To authenticate users, the Medrunner API uses Discord OAuth2. This allows users to log in using their Discord account, and then you can get the Medrunner API tokens related to the discord account of the user.
+
+To get your first refresh tokens, you need to start by redirecting the user to the discord login page. You should then pass your discord client id and the redirect_url parameter (that should be the url where your application is). Discord sends back a code as a get request to the CALLBACK_URL provided.
+
+```ts
+window.location.replace(
+  `https://discord.com/oauth2/authorize?client_id=${DISCORD_CLIENT_ID}&scope=identify&response_type=code&redirect_uri=${CALLBACK_URL}/auth/register`,
+);
+```
+
+Once you have the discord code, you can exchange it for the refresh token by making a POST request to the Medrunner API at the `/auth/signin` endpoint with the discord code as the `code` parameter, and the CALLBACK_URL you used in the discord request as the `redirectUri` parameter.
+
+```ts
+await axios.get(`https://api.medrunner.space/auth/signin?code=${DISCORD_CODE}&redirectUri=${CALLBACK_URL}/auth`);
+```
+
+::: warning
+Make sure to use the exact same CALLBACK_URL you used in the discord request, otherwise the request will fail.
+:::
+
+To authenticate API calls, you now need to exchange it for an access token, follow the [Authenticate API calls](/guides/authentication#authenticate-api-calls) section for detailed steps.
+
+## Authenticate API calls
+
+### Use the TypeScript client
+
+The easiest way to authenticate API calls is to use the TypeScript client provided by Medrunner. The client will handle the authentication for you, so you don't have to worry about it.
+
+To use the client, you need to initialize it with your API key or refresh token:
+
+```ts
+import MedrunnerApiClient from "@medrunner/api-client";
+
+const apiConfig = {
+  baseUrl: "https://api.medrunner.space",
+  token: "YOUR_REFRESH_TOKEN",
+};
+
+const refreshCallback = async newTokens => {
+  localStorage.setItem("refreshToken", newTokens.refreshToken);
+};
```

Comment:
when using API tokens, this is unnecessary as the refresh token returned is just the API token again

Reply:
Removed mentions of the refresh callback in 1289ab2e23a0b185962533363eaa925ef6f50cb1